templates/default/start-nm.sh.erb

```diff
@@ -5,6 +5,20 @@ h=$(hostname)
 bin=$(dirname "${BASH_SOURCE-$0}")
 bin=$(cd "$bin"; pwd)
 
+if [ <%= node['install']['localhost'] %> == "true" ] ; then
+    systemctl status nodemanager
+    if [ $? -ne 0 ] ; then
+	num_gpus=$(nvidia-smi -L | wc -l)
+	num_yarn_site=$(grep -C 2 '>yarn.nodemanager.resource.gpus<' yarn-site.xml | grep value | sed 's/<value>//' | sed 's/<\/value>//' | tr -d '[:space:]')
+
+	if [ $num_gpus -ne $num_yarn_site ] ; then
+	    # the nodemanager will recreate the correct number of cgroups. No cgroups should be running when this is executed
+	    rmdir /sys/fs/cgroup/devices/hops-yarn
```

Comment:
This script runs as user 'yarn' Does it have permission to delete this directory?